### PR TITLE
Add UnjailValidatorStateChangeMetadata

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* **#1048** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1048?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1047** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1047?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `feature/proof-of-stake`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1047?utm_source=stack-comment).

File: lib/state_change_metadata.go [hunk 1]
```diff
@@ -595,3 +596,41 @@ func (stakeRewardSCM *StakeRewardStateChangeMetadata) GetVersionByte(blockHeight
...
 	return EncoderTypeStakeRewardStateChangeMetadata
 }
+
+type UnjailValidatorStateChangeMetadata struct {
+	ValidatorPKID         *PKID
+	JailedAtEpochNumber   uint64
+	UnjailedAtEpochNumber uint64
+}
+
+func (unjailValidatorSCM *UnjailValidatorStateChangeMetadata) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
```
Comment (tholonious) [inline lib/state_change_metadata.go:606]: Maybe just shorten this to `metadata` for conciseness?
```suggestion
func (metadata *UnjailValidatorStateChangeMetadata) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
```

File: lib/state_change_metadata.go [hunk 2]
```diff
@@ -595,3 +596,41 @@ func (stakeRewardSCM *StakeRewardStateChangeMetadata) GetVersionByte(blockHeight
...
+	var data []byte
+	data = append(data, EncodeToBytes(blockHeight, unjailValidatorSCM.ValidatorPKID, skipMetadata...)...)
+	data = append(data, UintToBuf(unjailValidatorSCM.JailedAtEpochNumber)...)
+	data = append(data, UintToBuf(unjailValidatorSCM.UnjailedAtEpochNumber)...)
+	return data
+}
+
+func (unjailValidatorSCM *UnjailValidatorStateChangeMetadata) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
```
Comment (tholonious) [inline lib/state_change_metadata.go:614]:
```suggestion
func (metadata *UnjailValidatorStateChangeMetadata) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
```

File: lib/state_change_metadata.go [hunk 3]
```diff
@@ -595,3 +596,41 @@ func (stakeRewardSCM *StakeRewardStateChangeMetadata) GetVersionByte(blockHeight
...
+		return errors.Wrapf(err, "UnjailValidatorStateChangeMetadata.Decode: Problem reading JailedAtEpochNumber")
+	}
+	unjailValidatorSCM.UnjailedAtEpochNumber, err = ReadUvarint(rr)
+	if err != nil {
+		return errors.Wrapf(err, "UnjailValidatorStateChangeMetadata.Decode: Problem reading UnjailedAtEpochNumber")
+	}
+	return nil
+}
+
+func (unjailValidatorSCM *UnjailValidatorStateChangeMetadata) GetVersionByte(blockHeight uint64) byte {
```
Comment (tholonious) [inline lib/state_change_metadata.go:630]:
```suggestion
func (metadata *UnjailValidatorStateChangeMetadata) GetVersionByte(blockHeight uint64) byte {
```

File: lib/state_change_metadata.go [hunk 4]
```diff
@@ -595,3 +596,41 @@ func (stakeRewardSCM *StakeRewardStateChangeMetadata) GetVersionByte(blockHeight
...
+	return 0
+}
+
+func (unjailValidatorSCM *UnjailValidatorStateChangeMetadata) GetEncoderType() EncoderType {
```
Comment (tholonious) [inline lib/state_change_metadata.go:634]:
```suggestion
func (metadata *UnjailValidatorStateChangeMetadata) GetEncoderType() EncoderType {
```

**Review by tholonious — APPROVED** [review]

**Comment by lazynina** [comment]
### Merge activity

* **Feb 20, 1:23 PM EST**: @lazynina started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1047).
